diff --git a/modules/customer/includes/commerce_customer.checkout_pane.inc b/modules/customer/includes/commerce_customer.checkout_pane.inc
index cd50dad..257477e 100644
--- a/modules/customer/includes/commerce_customer.checkout_pane.inc
+++ b/modules/customer/includes/commerce_customer.checkout_pane.inc
@@ -206,7 +206,7 @@ function commerce_customer_profile_pane_checkout_form($form, &$form_state, $chec
     foreach (element_children($pane_form[$field_name][$langcode]) as $delta) {
       // Don't mess with the "Add another item" button that could be present.
       if ($pane_form[$field_name][$langcode][$delta]['#type'] != 'submit') {
-        $pane_form[$field_name][$langcode][$delta]['#type'] = 'container';
+        $pane_form[$field_name][$langcode][$delta]['#inline'] = TRUE;
       }
     }
   }
